File: NSKButtonObjC/NSKButtonObjCTV/TVViewController.m
//
//  TVViewController.m
//  NSKButtonObjCTV
//
//  Created by NSSimpleApps on 21.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "TVViewController.h"
#import "NSKButton.h"

@interface TVViewController ()

@property (weak, nonatomic) IBOutlet UIButton *nskButton;

@property (strong, nonatomic) NSArray<NSString *> *buttonTitles;

@end

@implementation TVViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    self.buttonTitles = @[@"Default", @"Right", @"Top", @"Bottom"];
    
    self.nskButton.layer.borderColor = [UIColor blueColor].CGColor;
    self.nskButton.layer.borderWidth = 1;
    [self.nskButton setTitle:self.buttonTitles.firstObject forState:UIControlStateNormal];
}

- (IBAction)changeImagePosition:(NSKButton *)sender {
    
    if (sender.nskImagePosition == NSKImagePositionBottom) {
        
        sender.nskImagePosition = NSKImagePositionDefault;
        
    } else {
        
        sender.nskImagePosition++;
    }
    
    [sender setTitle:self.buttonTitles[sender.nskImagePosition] forState:UIControlStateNormal];
}

@end

File: NSKButtonSwift/NSKButtonSwiftTV/TVViewController.swift
//
//  ViewController.swift
//  NSKButtonSwiftTV
//
//  Created by NSSimpleApps on 21.08.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import UIKit

class TVViewController: UIViewController {
    
    @IBOutlet weak var nskButton: NSKButton!
    
    private var buttonTitles = ["Default", "Right", "Top", "Bottom"]
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        self.nskButton.layer.borderColor =
            UIColor(red: 0, green: CGFloat(122)/255, blue: 1, alpha: 1).CGColor
        self.nskButton.layer.borderWidth = 1
        
        self.nskButton.setTitle(self.buttonTitles.first, forState: .Normal)
    }
    
    @IBAction func changeImagePosition(sender: NSKButton) {
        
        if (sender.nskImagePosition == .Bottom) {
            
            sender.nskImagePosition = .Default
            
        } else {
            
            let rawValue = sender.nskImagePosition.rawValue
            
            sender.nskImagePosition = NSKImagePosition(rawValue: rawValue + 1)!
        }
        
        sender.setTitle(self.buttonTitles[sender.nskImagePosition.rawValue], forState: .Normal)
    }
}

